app/assets/javascripts/discourse/app/routes/build-category-route.js

```diff
@@ -153,50 +126,17 @@ class AbstractCategoryRoute extends DiscourseRoute {
   }
 
   setupController(controller, model) {
-    const topics = this.topics,
-      category = model.category;
+    super.setupController(...arguments);
+    controller.bulkSelectHelper.clear();
+    this.searchService.searchContext = model.category.get("searchContext");
 
-    let topicOpts = {
-      model: topics,
-      category,
-      period:
-        topics.get("for_period") ||
-        (model.modelParams && model.modelParams.period),
-      noSubcategories: this.routeConfig && !!this.routeConfig.no_subcategories,
-      expandAllPinned: true,
-    };
-
-    const p = category.get("params");
-    if (p && Object.keys(p).length) {
-      if (p.order !== undefined) {
-        topicOpts.order = p.order;
-      }
-      if (p.ascending !== undefined) {
-        topicOpts.ascending = p.ascending;
-      }
+    const p = model.category.params;
+    if (p?.order !== undefined) {
+      controller.order = p.order;
```

Comment:
Are those props (`order` and `ascending` below) tracked? Or are they not but it does not matter to use `set()` here? 🙂

Reply:
They are tracked because they're in the queryParams list and then tracked-ified via

https://github.com/discourse/discourse/blob/c5b3d7b2f97757c3ed61a1f48525fec5f8968ecb/app/assets/javascripts/discourse/app/controllers/discovery/list.js#L62-L64